### DISCUSSION TITLE
Prometheus/targets页面 不显示kube-controller-manager kube-scheduler

### DISCUSSION THREAD

**Discussion by logwhee** [prometheus-operator/kube-prometheus#1004] [Q&A]
Prometheus/targets页面 不显示kube-controller-manager kube-scheduler 
请问怎么处理呀？？
![微信截图_20210219104756](https://user-images.githubusercontent.com/73780227/108450323-f9212e80-729f-11eb-98d5-9387b08d02c6.png)

**Comment by Atharex** [comment]
I guess the OP means the kube controller manager and scheduler are not working?

For me that is the case. The kube-prometheus installation picks up the kubelet and api server metrics, but not for the controller manager and scheduler. Alertmanager is showing these 2 warnings:

```
name: KubeControllerManagerDown
expr: absent(up{job="kube-controller-manager"} == 1)
for: 15m
labels:
severity: critical
annotations:
description: KubeControllerManager has disappeared from Prometheus target discovery.
runbook_url: https://github.com/prometheus-operator/kube-prometheus/wiki/kubecontrollermanagerdown
summary: Target disappeared from Prometheus target discovery.
```

```
KubeSchedulerDown (1 active)

name: KubeSchedulerDown
expr: absent(up{job="kube-scheduler"} == 1)
for: 15m
labels:
severity: critical
annotations:
description: KubeScheduler has disappeared from Prometheus target discovery.
runbook_url: https://github.com/prometheus-operator/kube-prometheus/wiki/kubeschedulerdown
summary: Target disappeared from Prometheus target discovery.
```

Is this a known issue?

**Comment by huangweikuna** [comment]
- 查看kube-controller-manager kube-scheduler对应的service
```kubectl get svc -n kube-system```
如果不存在则创建。
- 找到ServiceMonitor，查看Service选择标签、名称、协议是否匹配
- 如果是托管到第三方服务，则没有这两个target

希望对你有帮助。

**Comment by starcraft66** [comment]
@Atharex I'm experiencing the same issue with a fresh install of release 0.7 on a bare metal k8s 1.20 cluster deployed with kubespray.

**Comment by paulfantom** [comment]
You need to enable gathering metrics from kube-scheduler and kube-controller-manager or disable those alerts.

This should be covered in our docs: https://github.com/prometheus-operator/kube-prometheus#cluster-creation-tools
